rocAL/rocAL/source/coco_meta_data_reader_key_points.cpp

```diff
@@ -0,0 +1,396 @@
+                // label is not person (label !=1)
+                // joint_sum <= 0
+                // is_crowd==1
+                if (label != 1 || joint_sum <= 0 || is_crowd == 1)
+                {
+                    std::fill(std::begin(box_center), std::end(box_center), 0.0);
+                    std::fill(std::begin(box_scale), std::end(box_scale), 0.0);
+                    continue;
+                }
+
+                // Validate bbox values
+                float x1, y1, x2, y2;
+                x1 = std::max(box_center[0] , 0.0f);
+                y1 = std::max(box_center[1] , 0.0f);
+                float box_w = std::max(box_scale[0] - 1 , 0.0f);
+                float box_h = std::max(box_scale[1] - 1 , 0.0f);
+                x2 = std::min((float)image_size.w - 1 ,  x1 + box_w);
+                y2 = std::min((float)image_size.h - 1 , y1 + box_h);
+
+                // check area
+                if (area > 0 && x2 >= x1 && y2 >= y1)
+                {
+                    box_center[0] = x1;
+                    box_center[1] = y1;
+                    box_scale[0] = x2 - x1;
+                    box_scale[1] = y2 - y1;
+                }
+
+                // Convert from xywh to center,scale
+                box_center[0] += (0.5 * box_scale[0]);
+                box_center[1] += (0.5 * box_scale[1]);
+
+                if (box_scale[0] > aspect_ratio * box_scale[1])
+                {
+                    box_scale[1] = box_scale[0] * inverse_aspect_ratio * inverse_pixel_std;
+                    box_scale[0] = box_scale[0] * inverse_pixel_std;
+                }
+                else if (box_scale[0] < aspect_ratio * box_scale[1])
+                {
+                    box_scale[0] = box_scale[1] * aspect_ratio * inverse_pixel_std;
+                    box_scale[1] = box_scale[1] * inverse_pixel_std;
+                }
+
+                if (box_center[0] != -1)
+                {
+                    box_scale[0] = SCALE_CONSTANT_CS * box_scale[0];
+                    box_scale[1] = SCALE_CONSTANT_CS * box_scale[1];
+                }
+
+                // Convert raw keypoint values to Joints,Joint Visibilities - Clip the visibilities to range [0,1]
+                std::vector<std::vector<float>> joints(NUMBER_OF_JOINTS),joints_visibility(NUMBER_OF_JOINTS);
+                unsigned int j = 0;
+                for (unsigned int i = 0; i < NUMBER_OF_JOINTS; i++)
+                {
+                    joints[i].push_back(keypoint[j]);
+                    joints[i].push_back(keypoint[j + 1]);
+                    if( keypoint[j + 2] > 1.0)
+                    {
+                        keypoint[j + 2] = 1.0;
+                    }
+                    joints_visibility[i].push_back(keypoint[j + 2]);
+                    joints_visibility[i].push_back(keypoint[j + 2]);
+                    j = j + 3;
+                }
+
+                // Add values to joints_data structure
```

Comment:
Storing lots of meta_data like this for the entire data_set might use lot of memory. Please evaluate how much memory is used for storing joints data for the entire coco dataset?

Reply:
JointsData needs - 320 bytes of memory per annotation
Valid keypoint annotations in entire coco dataset - 156160 (train+val)
It needs nearly 50MB of memory for entire coco dataset